### PR TITLE
Reformat/simplify

### PR DESCRIPTION
**Pull request by quinnabrvau**
## Refactor
I was just reviewing the code and noticed that there were a bunch of places where the same code was written repeatably. So I started by simplifying the code by putting repeated calls into functions. I was then noticing that there wasn't anyway to test the changes that I was making. So I reorganized the code into a source file and added a test folder. 

## Test Code
I first made a simple test program to check `api.py` this functionality is critical to the project and was an easy place to start. From there I went to add tests running `DynamoDB` locally. So far, I can connect to the database from my test code. Next I need to create a table.

## Side project
(maybe this should be separate pr but I was playing with it) support for numerical rank searches such as`!bgrank 200 NA`. I will add tests for this once I have the test code working :)

## How to test:
1. build with `docker-compose build`
2. run with `docker-compose run app-node`

### CODE REVIEW

File: src/discordBot.py [hunk 1]
```diff
@@ -24,84 +25,30 @@ def getEmbedObject(text, player, command):
...
-        args = ['lii', region]
-
-    # Handle if a rank is passed in with a region
-    # if len(args) > 1:
-    #     args[0] = get_tag_from_rank(args[0], args[1])
-
-    response = removeTwitchEmotes(leaderboardBot.getRankText(*args))
-
+def call(ctx, func, name, *args):
+    response = removeTwitchEmotes(func(*args))
```
Comment (JimLiu0) [inline src/discordBot.py:31]: May not need to removeTwitchEmotes twice as its called in the ctx.send call below.

File: src/leaderboardBot.py [hunk 2]
```diff
@@ -21,23 +21,53 @@
...
 
-    def __init__(self):
-        self.db = boto3.resource('dynamodb', aws_access_key_id=os.environ['AWS_ACCESS_KEY_ID'], aws_secret_access_key=os.environ['AWS_ACCESS_KEY'], region_name=os.environ['REGION'])
+    def __init__(self, url=None):
+        self.db = None
+        if url is None:
+            self.db = boto3.resource('dynamodb',
+                aws_access_key_id=os.environ['AWS_ACCESS_KEY_ID'],
+                aws_secret_access_key=os.environ['AWS_SECRET_ACCESS_KEY'],
+                region_name=os.environ['REGION'])
```
Comment (JimLiu0) [inline src/leaderboardBot.py]: Can maybe combine these.

Comment (quinnabrvau) [reply src/leaderboardBot.py]: I was thinking about using kwargs but I was having issues was I moved on

File: src/discordBot.py (hunk 1 above)
Comment (quinnabrvau) [reply src/discordBot.py:31]: this was one of the places I wasn't sure on, do I need to make this an `async def`. Agree this only needs to be called once.

**Pull request merged**